# STEAM DATASET CLEANING PROCESS

In this part of the project I am going to clean a dataset that has information about Steam. This dataset is publicly available in Kaggle(link:https://www.kaggle.com/datasets/nikdavis/steam-store-games). The original state of the data is not very useful. We have missing values & columns with bad formatting. 

The roadmap we will follow to clean the dataset is going to be the following:
- General overview: have a general look of the current state of the dataset
- Missing values: manage the different missing values. In some cases we may found possible solutions, in other cases we will just remove those values.
- Duplicates: look for possibles duplicates
- Columns formatting and creation: there are some columns with a format that makes it difficult to work with the data. We will handle those column and transform them into viable data. We will also create new column based on the original dataset.


### GENERAL OVERVIEW

In [ ]:
import pandas as pd
import  numpy as np

In [2]:
df = pd.read_csv('games.csv')

In [3]:
df.head()

AppID                   Name  Release date Estimated owners  Peak CCU  \
0    20200       Galactic Bowling  Oct 21, 2008        0 - 20000         0   
1   655370           Train Bandit  Oct 12, 2017        0 - 20000         0   
2  1732930           Jolt Project  Nov 17, 2021        0 - 20000         0   
3  1355720               Henosis™  Jul 23, 2020        0 - 20000         0   
4  1139950  Two Weeks in Painland   Feb 3, 2020        0 - 20000         0   

   Required age  Price  DLC count  \
0             0  19.99          0   
1             0   0.99          0   
2             0   4.99          0   
3             0   5.99          0   
4             0   0.00          0   

                                      About the game  \
0  Galactic Bowling is an exaggerated and stylize...   
1  THE LAW!! Looks to be a showdown atop a train....   
2  Jolt Project: The army now has a new robotics ...   
3  HENOSIS™ is a mysterious 2D Platform Puzzler w...   
4  ABOUT THE GAME Play as a hacker who has arrang...   

                                 Supported languages  ...  \
0                                        ['English']  ...   
1  ['English', 'French', 'Italian', 'German', 'Sp...  ...   
2                 ['English', 'Portuguese - Brazil']  ...   
3  ['English', 'French', 'Italian', 'German', 'Sp...  ...   
4                     ['English', 'Spanish - Spain']  ...   

  Average playtime two weeks Median playtime forever  \
0                          0                       0   
1                          0                       0   
2                          0                       0   
3                          0                       0   
4                          0                       0   

  Median playtime two weeks             Developers             Publishers  \
0                         0  Perpetual FX Creative  Perpetual FX Creative   
1                         0           Rusty Moyher           Wild Rooster   
2                         0          Campião Games          Campião Games   
3                         0      Odd Critter Games      Odd Critter Games   
4                         0          Unusual Games          Unusual Games   

                                          Categories  \
0  Single-player,Multi-player,Steam Achievements,...   
1  Single-player,Steam Achievements,Full controll...   
2                                      Single-player   
3              Single-player,Full controller support   
4                   Single-player,Steam Achievements   

                            Genres  \
0              Casual,Indie,Sports   
1                     Action,Indie   
2  Action,Adventure,Indie,Strategy   
3           Adventure,Casual,Indie   
4                  Adventure,Indie   

                                                Tags  \
0                        Indie,Casual,Sports,Bowling   
1  Indie,Action,Pixel Graphics,2D,Retro,Arcade,Sc...   
2                                                NaN   
3  2D Platformer,Atmospheric,Surreal,Mystery,Puzz...   
4  Indie,Adventure,Nudity,Violent,Sexual Content,...   

                                         Screenshots  \
0  https://cdn.akamai.steamstatic.com/steam/apps/...   
1  https://cdn.akamai.steamstatic.com/steam/apps/...   
2  https://cdn.akamai.steamstatic.com/steam/apps/...   
3  https://cdn.akamai.steamstatic.com/steam/apps/...   
4  https://cdn.akamai.steamstatic.com/steam/apps/...   

                                              Movies  
0  http://cdn.akamai.steamstatic.com/steam/apps/2...  
1  http://cdn.akamai.steamstatic.com/steam/apps/2...  
2  http://cdn.akamai.steamstatic.com/steam/apps/2...  
3  http://cdn.akamai.steamstatic.com/steam/apps/2...  
4  http://cdn.akamai.steamstatic.com/steam/apps/2...  

[5 rows x 39 columns]

In [4]:
df.shape

(68223, 39)

In [5]:
df['Estimated owners'].unique()

array(['0 - 20000', '50000 - 100000', '20000 - 50000', '200000 - 500000',
       '100000 - 200000', '2000000 - 5000000', '0 - 0',
       '500000 - 1000000', '1000000 - 2000000', '20000000 - 50000000',
       '5000000 - 10000000', '10000000 - 20000000',
       '50000000 - 100000000', '100000000 - 200000000'], dtype=object)

In [6]:
group = df.groupby('Estimated owners').agg({'Name':'count'})
group

Name
Estimated owners            
0 - 0                   5776
0 - 20000              45401
100000 - 200000         2406
1000000 - 2000000        500
10000000 - 20000000       38
100000000 - 200000000      1
20000 - 50000           7187
200000 - 500000         2030
2000000 - 5000000        318
20000000 - 50000000       21
50000 - 100000          3582
500000 - 1000000         873
5000000 - 10000000        85
50000000 - 100000000       4

In [7]:
df.columns

Index(['AppID', 'Name', 'Release date', 'Estimated owners', 'Peak CCU',
       'Required age', 'Price', 'DLC count', 'About the game',
       'Supported languages', 'Full audio languages', 'Reviews',
       'Header image', 'Website', 'Support url', 'Support email', 'Windows',
       'Mac', 'Linux', 'Metacritic score', 'Metacritic url', 'User score',
       'Positive', 'Negative', 'Score rank', 'Achievements', 'Recommendations',
       'Notes', 'Average playtime forever', 'Average playtime two weeks',
       'Median playtime forever', 'Median playtime two weeks', 'Developers',
       'Publishers', 'Categories', 'Genres', 'Tags', 'Screenshots', 'Movies'],
      dtype='object')

In [8]:
columns = ['AppID', 'Name', 'Release date', 'Estimated owners', 'Peak CCU',
       'Price','Metacritic score', 'User score',
       'Positive', 'Negative','Recommendations',
       'Average playtime forever', 
       'Median playtime forever', 'Developers',
       'Publishers', 'Categories', 'Genres', 'Tags']

games = df[(columns)]

In [10]:
games[['Name', 'Developers', 'Publishers','Estimated owners', 'Release date']]

Name                     Developers  \
0                            Galactic Bowling          Perpetual FX Creative   
1                                Train Bandit                   Rusty Moyher   
2                                Jolt Project                  Campião Games   
3                                    Henosis™              Odd Critter Games   
4                       Two Weeks in Painland                  Unusual Games   
...                                       ...                            ...   
68218                 The Starving Tournament                     GJTP Games   
68219     Voice of the Wizard by Brett Farkas                   Brett Farkas   
68220                              Blockworks                    Elastic Sea   
68221  Stinky and Loof in Wonderland Playtest                            NaN   
68222                                  PINHI!  William Wheate,Charlie Wheate   

                          Publishers Estimated owners  Release date  
0              Perpetual FX Creative        0 - 20000  Oct 21, 2008  
1                       Wild Rooster        0 - 20000  Oct 12, 2017  
2                      Campião Games        0 - 20000  Nov 17, 2021  
3                  Odd Critter Games        0 - 20000  Jul 23, 2020  
4                      Unusual Games        0 - 20000   Feb 3, 2020  
...                              ...              ...           ...  
68218                     GJTP Games            0 - 0   Jan 6, 2023  
68219                   Brett Farkas        0 - 20000   Jan 5, 2023  
68220                    Elastic Sea        0 - 20000   Jan 7, 2023  
68221                            NaN            0 - 0   Jan 8, 2023  
68222  William Wheate,Charlie Wheate            0 - 0   Jan 8, 2023  

[68223 rows x 5 columns]

In [11]:
games['Developers'].unique

<bound method Series.unique of 0                Perpetual FX Creative
1                         Rusty Moyher
2                        Campião Games
3                    Odd Critter Games
4                        Unusual Games
                     ...              
68218                       GJTP Games
68219                     Brett Farkas
68220                      Elastic Sea
68221                              NaN
68222    William Wheate,Charlie Wheate
Name: Developers, Length: 68223, dtype: object>

We are going to make the first change in the original dataset. The dataset has data from 2003 to 2022. I decided to focus my analysis in the 2010 to 2021 period. I think that the current state of the gaming industry has nothing to do with what the industry was at the beginning of the century. The aim of the project is to find out what makes a game a success. To complete these objective, I think the best option is to look how games have performed in the last 10 years.



In [13]:
#We are going to analyze the period from 2010-2021, let's filter the data
games['Release date'] = pd.to_datetime(games['Release date'], errors = 'coerce')

C:\Users\manum\AppData\Local\Temp\ipykernel_4496\707575881.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  games['Release date'] = pd.to_datetime(games['Release date'], errors = 'coerce')


In [14]:
games = games[(games['Release date'] < '2022-01-01') & (games['Release date'] > '2009-12-31')]


In [15]:
games.head()

AppID                   Name Release date Estimated owners  Peak CCU  \
1   655370           Train Bandit   2017-10-12        0 - 20000         0   
2  1732930           Jolt Project   2021-11-17        0 - 20000         0   
3  1355720               Henosis™   2020-07-23        0 - 20000         0   
4  1139950  Two Weeks in Painland   2020-02-03        0 - 20000         0   
5  1469160         Wartune Reborn   2021-02-26   50000 - 100000        68   

   Price  Metacritic score  User score  Positive  Negative  Recommendations  \
1   0.99                 0           0        53         5                0   
2   4.99                 0           0         0         0                0   
3   5.99                 0           0         3         0                0   
4   0.00                 0           0        50         8                0   
5   0.00                 0           0        87        49                0   

   Average playtime forever  Median playtime forever         Developers  \
1                         0                        0       Rusty Moyher   
2                         0                        0      Campião Games   
3                         0                        0  Odd Critter Games   
4                         0                        0      Unusual Games   
5                         0                        0              7Road   

          Publishers                                         Categories  \
1       Wild Rooster  Single-player,Steam Achievements,Full controll...   
2      Campião Games                                      Single-player   
3  Odd Critter Games              Single-player,Full controller support   
4      Unusual Games                   Single-player,Steam Achievements   
5              7Road  Single-player,Multi-player,MMO,PvP,Online PvP,...   

                                              Genres  \
1                                       Action,Indie   
2                    Action,Adventure,Indie,Strategy   
3                             Adventure,Casual,Indie   
4                                    Adventure,Indie   
5  Adventure,Casual,Free to Play,Massively Multip...   

                                                Tags  
1  Indie,Action,Pixel Graphics,2D,Retro,Arcade,Sc...  
2                                                NaN  
3  2D Platformer,Atmospheric,Surreal,Mystery,Puzz...  
4  Indie,Adventure,Nudity,Violent,Sexual Content,...  
5  Turn-Based Combat,Massively Multiplayer,Multip...

### NULL VALUES

In [16]:
games.isna().sum()

AppID                          0
Name                           1
Release date                   0
Estimated owners               0
Peak CCU                       0
Price                          0
Metacritic score               0
User score                     0
Positive                       0
Negative                       0
Recommendations                0
Average playtime forever       0
Median playtime forever        0
Developers                   974
Publishers                  1184
Categories                  1782
Genres                       948
Tags                        5248
dtype: int64

In [17]:
games.loc[games['Name'].isna()]
#remove
games = games.drop(games.loc[games['Name'].isna()].index)

In [18]:
games.isna().sum()

AppID                          0
Name                           0
Release date                   0
Estimated owners               0
Peak CCU                       0
Price                          0
Metacritic score               0
User score                     0
Positive                       0
Negative                       0
Recommendations                0
Average playtime forever       0
Median playtime forever        0
Developers                   973
Publishers                  1183
Categories                  1781
Genres                       947
Tags                        5247
dtype: int64

In [19]:
games.loc[games['Developers'].isna()]

AppID                                           Name Release date  \
214    1688630                      Emperial Knights Playtest   2021-11-14   
220    1478660                         Slotracers VR Playtest   2020-12-17   
291    1613340          Pirates of the Asteroid Belt Playtest   2021-04-23   
300    1743770                             Propnight Playtest   2021-10-15   
315    1774150                      Stack Gun Heroes Playtest   2021-10-08   
...        ...                                            ...          ...   
61164  1822620                         Trash Sailors Playtest   2021-11-17   
61428  1857430                        Eternal Exodus Playtest   2021-12-23   
61445  1691520                  Erannorth Chronicles Playtest   2021-08-29   
61641  1850720  La Maledizione dell'Uccello Serpente Playtest   2021-12-17   
61649  1801330                         Blade Crusade Playtest   2021-11-26   

      Estimated owners  Peak CCU  Price  Metacritic score  User score  \
214              0 - 0         0    0.0                 0           0   
220              0 - 0         0    0.0                 0           0   
291              0 - 0         0    0.0                 0           0   
300              0 - 0         0    0.0                 0           0   
315              0 - 0         0    0.0                 0           0   
...                ...       ...    ...               ...         ...   
61164            0 - 0         0    0.0                 0           0   
61428            0 - 0         0    0.0                 0           0   
61445            0 - 0         0    0.0                 0           0   
61641            0 - 0         0    0.0                 0           0   
61649            0 - 0         0    0.0                 0           0   

       Positive  Negative  Recommendations  Average playtime forever  \
214           0         0                0                         0   
220           0         0                0                         0   
291           0         0                0                         0   
300           0         0                0                         0   
315           0         0                0                         0   
...         ...       ...              ...                       ...   
61164         0         0                0                         0   
61428         0         0                0                         0   
61445         0         0                0                         0   
61641         0         0                0                         0   
61649         0         0                0                         0   

       Median playtime forever Developers Publishers Categories Genres Tags  
214                          0        NaN        NaN        NaN    NaN  NaN  
220                          0        NaN        NaN        NaN    NaN  NaN  
291                          0        NaN        NaN        NaN    NaN  NaN  
300                          0        NaN        NaN        NaN    NaN  NaN  
315                          0        NaN        NaN        NaN    NaN  NaN  
...                        ...        ...        ...        ...    ...  ...  
61164                        0        NaN        NaN        NaN    NaN  NaN  
61428                        0        NaN        NaN        NaN    NaN  NaN  
61445                        0        NaN        NaN        NaN    NaN  NaN  
61641                        0        NaN        NaN        NaN    NaN  NaN  
61649                        0        NaN        NaN        NaN    NaN  NaN  

[973 rows x 18 columns]

In [20]:
#I found that there are 289 games that the developer is mentioned but the publisher is not mentioned
#so I assigned the publisher to be the same as the developer
for index, row in games.iterrows():
    if(pd.isnull(row['Publishers'])):
        if(pd.isnull(row['Developers'])):
            continue
        else:
            games.at[index, 'Publishers'] = games.at[index, 'Developers']

In [21]:
#Same but with devs
for index, row in games.iterrows():
    if (pd.isnull(row['Developers'])):
        if (pd.isnull(row['Publishers'])):
            continue
        else:
            games.at[index, 'Developers'] = games.at[index, 'Publishers']

In [22]:
games.isna().sum()

AppID                          0
Name                           0
Release date                   0
Estimated owners               0
Peak CCU                       0
Price                          0
Metacritic score               0
User score                     0
Positive                       0
Negative                       0
Recommendations                0
Average playtime forever       0
Median playtime forever        0
Developers                   918
Publishers                   918
Categories                  1781
Genres                       947
Tags                        5247
dtype: int64

In [23]:
games.loc[games['Developers'].isna()]

AppID                                           Name Release date  \
214    1688630                      Emperial Knights Playtest   2021-11-14   
220    1478660                         Slotracers VR Playtest   2020-12-17   
291    1613340          Pirates of the Asteroid Belt Playtest   2021-04-23   
300    1743770                             Propnight Playtest   2021-10-15   
315    1774150                      Stack Gun Heroes Playtest   2021-10-08   
...        ...                                            ...          ...   
61164  1822620                         Trash Sailors Playtest   2021-11-17   
61428  1857430                        Eternal Exodus Playtest   2021-12-23   
61445  1691520                  Erannorth Chronicles Playtest   2021-08-29   
61641  1850720  La Maledizione dell'Uccello Serpente Playtest   2021-12-17   
61649  1801330                         Blade Crusade Playtest   2021-11-26   

      Estimated owners  Peak CCU  Price  Metacritic score  User score  \
214              0 - 0         0    0.0                 0           0   
220              0 - 0         0    0.0                 0           0   
291              0 - 0         0    0.0                 0           0   
300              0 - 0         0    0.0                 0           0   
315              0 - 0         0    0.0                 0           0   
...                ...       ...    ...               ...         ...   
61164            0 - 0         0    0.0                 0           0   
61428            0 - 0         0    0.0                 0           0   
61445            0 - 0         0    0.0                 0           0   
61641            0 - 0         0    0.0                 0           0   
61649            0 - 0         0    0.0                 0           0   

       Positive  Negative  Recommendations  Average playtime forever  \
214           0         0                0                         0   
220           0         0                0                         0   
291           0         0                0                         0   
300           0         0                0                         0   
315           0         0                0                         0   
...         ...       ...              ...                       ...   
61164         0         0                0                         0   
61428         0         0                0                         0   
61445         0         0                0                         0   
61641         0         0                0                         0   
61649         0         0                0                         0   

       Median playtime forever Developers Publishers Categories Genres Tags  
214                          0        NaN        NaN        NaN    NaN  NaN  
220                          0        NaN        NaN        NaN    NaN  NaN  
291                          0        NaN        NaN        NaN    NaN  NaN  
300                          0        NaN        NaN        NaN    NaN  NaN  
315                          0        NaN        NaN        NaN    NaN  NaN  
...                        ...        ...        ...        ...    ...  ...  
61164                        0        NaN        NaN        NaN    NaN  NaN  
61428                        0        NaN        NaN        NaN    NaN  NaN  
61445                        0        NaN        NaN        NaN    NaN  NaN  
61641                        0        NaN        NaN        NaN    NaN  NaN  
61649                        0        NaN        NaN        NaN    NaN  NaN  

[918 rows x 18 columns]

In [24]:
games.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 53358 entries, 1 to 67973
Data columns (total 18 columns):
 #   Column                    Non-Null Count  Dtype         
---  ------                    --------------  -----         
 0   AppID                     53358 non-null  int64         
 1   Name                      53358 non-null  object        
 2   Release date              53358 non-null  datetime64[ns]
 3   Estimated owners          53358 non-null  object        
 4   Peak CCU                  53358 non-null  int64         
 5   Price                     53358 non-null  float64       
 6   Metacritic score          53358 non-null  int64         
 7   User score                53358 non-null  int64         
 8   Positive                  53358 non-null  int64         
 9   Negative                  53358 non-null  int64         
 10  Recommendations           53358 non-null  int64         
 11  Average playtime forever  53358 non-null  int64         
 12  Median playtime fo

In [25]:
contain_values = games[games['Name'].str.contains('Playtest')]
print(contain_values)

         AppID                                           Name Release date  \
214    1688630                      Emperial Knights Playtest   2021-11-14   
220    1478660                         Slotracers VR Playtest   2020-12-17   
291    1613340          Pirates of the Asteroid Belt Playtest   2021-04-23   
300    1743770                             Propnight Playtest   2021-10-15   
315    1774150                      Stack Gun Heroes Playtest   2021-10-08   
...        ...                                            ...          ...   
61164  1822620                         Trash Sailors Playtest   2021-11-17   
61428  1857430                        Eternal Exodus Playtest   2021-12-23   
61445  1691520                  Erannorth Chronicles Playtest   2021-08-29   
61641  1850720  La Maledizione dell'Uccello Serpente Playtest   2021-12-17   
61649  1801330                         Blade Crusade Playtest   2021-11-26   

      Estimated owners  Peak CCU  Price  Metacritic score  User

In [26]:
games = games.drop(games.loc[games['Developers'].isna()].index)
games = games.drop(games.loc[games['Publishers'].isna()].index)


In [27]:
games.isna().sum()

AppID                          0
Name                           0
Release date                   0
Estimated owners               0
Peak CCU                       0
Price                          0
Metacritic score               0
User score                     0
Positive                       0
Negative                       0
Recommendations                0
Average playtime forever       0
Median playtime forever        0
Developers                     0
Publishers                     0
Categories                   907
Genres                        73
Tags                        4331
dtype: int64

In [28]:
games.loc[games['Categories'].isna()]

AppID                                 Name Release date  \
31     1727670                   Home Office Tasker   2021-09-08   
145    1302840          Kooring VR Coding Adventure   2020-08-01   
359     417540              Gamefuel Driver Control   2015-12-10   
450     620050                              Start10   2017-05-11   
543     461550  Typing Instructor Platinum 21 - Mac   2017-02-10   
...        ...                                  ...          ...   
61395  1440450                           Metagroove   2020-11-20   
61502   830640                           DayZ Tools   2018-11-07   
61532  1702750                            idol Live   2021-08-23   
61569   925160       Sky Residences at Ice District   2019-02-08   
61666  1366800                          Crosshair X   2020-08-17   

      Estimated owners  Peak CCU  Price  Metacritic score  User score  \
31           0 - 20000         0   0.99                 0           0   
145          0 - 20000         0   8.49                 0           0   
359      20000 - 50000         0  29.99                 0           0   
450          0 - 20000         0   4.99                 0           0   
543          0 - 20000         0  29.99                 0           0   
...                ...       ...    ...               ...         ...   
61395        0 - 20000         0   0.00                 0           0   
61502        0 - 20000        46   0.00                 0           0   
61532        0 - 20000         0   2.99                 0           0   
61569        0 - 20000         0   0.00                 0           0   
61666  200000 - 500000      2514   3.99                 0           0   

       Positive  Negative  Recommendations  Average playtime forever  \
31           11         4                0                         0   
145           8         0                0                         0   
359           9         6                0                         0   
450          38        26                0                         1   
543           0         0                0                         0   
...         ...       ...              ...                       ...   
61395        28         9                0                         0   
61502       140        20              135                         0   
61532         1         0                0                         0   
61569         5         0                0                         0   
61666      3848       408             3933                      5292   

       Median playtime forever                                  Developers  \
31                           0                                    lonch.me   
145                          0                                  VRANI inc.   
359                          0                          Auslogics Software   
450                          1                                    Stardock   
543                          0                         Individual Software   
...                        ...                                         ...   
61395                        0                             Josh Presseisen   
61502                        0                         Bohemia Interactive   
61532                        0  Zhejiang Tuxing Zhiyu Technology Co., Ltd.   
61569                        0              Real Estate in Virtual Reality   
61666                     3042                          CenterPoint Gaming   

                                       Publishers Categories  \
31                                       lonch.me        NaN   
145                                    VRANI inc.        NaN   
359                              Console Classics        NaN   
450                                      Stardock        NaN   
543                           Individual Software        NaN   
...                                           ...        ...   
61395                         Crescent Moon Games        NaN   
6

In [29]:
games.isna().sum()

AppID                          0
Name                           0
Release date                   0
Estimated owners               0
Peak CCU                       0
Price                          0
Metacritic score               0
User score                     0
Positive                       0
Negative                       0
Recommendations                0
Average playtime forever       0
Median playtime forever        0
Developers                     0
Publishers                     0
Categories                   907
Genres                        73
Tags                        4331
dtype: int64

In [30]:
for index, row in games.iterrows():
    if(pd.isnull(row['Categories'])):
        games.at[index, 'Categories'] = 'no Category added'

In [31]:
games.loc[games['Genres'].isna()]

AppID                                               Name Release date  \
451    211208                                        VectorMan 2   2012-05-02   
1004   211200                                       Beyond Oasis   2012-05-02   
1346    71250                                 Sonic Adventure DX   2011-03-04   
3002   529640                            SteamVR Driver for FOVE   2019-01-11   
5542   675500  ACT IT OUT XL! A Game of Charades - Designed f...   2018-03-06   
...       ...                                                ...          ...   
59897  202530                   Sonic the Hedgehog 4 - Episode I   2012-01-19   
60641   34283                 Alex Kidd™ in the Enchanted Castle   2010-09-13   
60665  306510                          CRYENGINE - Sample Assets   2014-06-06   
61449  682910                                          No Lights   2017-08-07   
61502  830640                                         DayZ Tools   2018-11-07   

       Estimated owners  Peak CCU  Price  Metacritic score  User score  \
451               0 - 0         0   0.99                 0           0   
1004              0 - 0         0   0.99                 0           0   
1346   500000 - 1000000       115   7.99                 0           0   
3002          0 - 20000         0   0.00                 0           0   
5542          0 - 20000         0  11.99                 0           0   
...                 ...       ...    ...               ...         ...   
59897   200000 - 500000         8   9.99                 0           0   
60641         0 - 20000         0   0.99                 0           0   
60665         0 - 20000         0   0.00                 0           0   
61449    50000 - 100000         0   2.99                 0           0   
61502         0 - 20000        46   0.00                 0           0   

       Positive  Negative  Recommendations  Average playtime forever  \
451           0         0                0                         0   
1004          0         0                0                         0   
1346       9673      1097             8734                       360   
3002          2         0                0                         0   
5542         36        11                0                         0   
...         ...       ...              ...                       ...   
59897      1055      1439             1595                       125   
60641         5         4                0                         0   
60665         2         0                0                         0   
61449        57        54                0                         4   
61502       140        20              135                         0   

       Median playtime forever           Developers           Publishers  \
451                          0                 SEGA                 SEGA   
1004                         0                 SEGA                 SEGA   
1346                       126                 SEGA                 SEGA   
3002                         0           FOVE, Inc.           FOVE, Inc.   
5542                         0    Snap Finger Click    Snap Finger Click   
...                        ...                  ...                  ...   
59897                      100                 SEGA                 SEGA   
60641                        0                 SEGA                 SEGA   
60665                        0               Crytek               Crytek   
61449                        4             Cubamano            EasyGames   
61502                        0  Bohemia Interactive  Bohemia Interactive   

                                              Categories Genres  \
451                                        Single-player    NaN   
1004                                       Single-player    NaN   
1346            Single-player,Partial Controller Support    NaN   
3002                                   no Category added    NaN   
5542   Single-player,Multi-player

In [32]:
# dealing with 100 games left
for index, row in games.iterrows():
    if(pd.isnull(row['Genres'])):
        games.at[index, 'Genres'] = 'no Genres added'

In [33]:
games.isna().sum()

AppID                          0
Name                           0
Release date                   0
Estimated owners               0
Peak CCU                       0
Price                          0
Metacritic score               0
User score                     0
Positive                       0
Negative                       0
Recommendations                0
Average playtime forever       0
Median playtime forever        0
Developers                     0
Publishers                     0
Categories                     0
Genres                         0
Tags                        4331
dtype: int64

In [34]:
for index, row in games.iterrows():
    if(pd.isnull(row['Tags'])):
        games.at[index, 'Tags'] = games.at[index, 'Genres']

In [35]:
games.isna().sum()

AppID                       0
Name                        0
Release date                0
Estimated owners            0
Peak CCU                    0
Price                       0
Metacritic score            0
User score                  0
Positive                    0
Negative                    0
Recommendations             0
Average playtime forever    0
Median playtime forever     0
Developers                  0
Publishers                  0
Categories                  0
Genres                      0
Tags                        0
dtype: int64

All the missing values have been removed

### DUPLICATES

In [36]:
duplicates = games[games.duplicated()]
duplicates

Empty DataFrame
Columns: [AppID, Name, Release date, Estimated owners, Peak CCU, Price, Metacritic score, User score, Positive, Negative, Recommendations, Average playtime forever, Median playtime forever, Developers, Publishers, Categories, Genres, Tags]
Index: []

We have no duplicates

In [37]:
#No duplicates
games.sample(10)

AppID                                  Name Release date  \
19290  1248970                   Hentai Amazon Girls   2020-03-08   
55101  1387060                 Tale of an Apocalypse   2020-10-14   
34096   765320           Planetarium 2 - Zen Odyssey   2018-01-03   
57398  1174690                           love wish 2   2020-06-19   
26277  1171130                     Scraper: Gauntlet   2020-02-20   
44196   418240  Shadow Tactics: Blades of the Shogun   2016-12-06   
13127   528720                                   Shu   2016-10-04   
32871   463450                          Bipolar Game   2016-04-19   
33790   395880                             Sky Tower   2016-04-25   
59145  1073310                  Hentai Zodiac Puzzle   2019-05-10   

        Estimated owners  Peak CCU  Price  Metacritic score  User score  \
19290          0 - 20000         0   0.99                 0           0   
55101          0 - 20000         0  14.99                 0           0   
34096          0 - 20000         0   1.99                 0           0   
57398      20000 - 50000         2   1.99                 0           0   
26277          0 - 20000         0  14.99                 0           0   
44196  1000000 - 2000000       322  39.99                85           0   
13127      20000 - 50000         0   9.99                85           0   
32871          0 - 20000         0   9.99                 0           0   
33790          0 - 20000         0   4.99                 0           0   
59145          0 - 20000         2   0.99                 0           0   

       Positive  Negative  Recommendations  Average playtime forever  \
19290         0         0                0                         0   
55101         3         4                0                         0   
34096         7         4                0                         0   
57398      1015       247             1366                        64   
26277        25         1                0                         0   
44196     30558      1192            25511                      2264   
13127       127        33                0                       338   
32871         6         0                0                         0   
33790         8         4                0                         0   
59145         0         0              160                        64   

       Median playtime forever             Developers              Publishers  \
19290                        0            Seito Games             Seito Games   
55101                        0       Steve Suetterlin        Steve Suetterlin   
34096                        0           Ghulam Jewel            Ghulam Jewel   
57398                       60               Double W             Toffee Cafe   
26277                        0         Labrodex, Inc.          Labrodex, Inc.   
44196                      231           Mimimi Games  Daedalic Entertainment   
13127                      338  Coatsink,Secret Lunch                Coatsink   
32871                        0             Jason Free              Jason Free   
33790                        0              Animootor               Animootor   
59145                       64         Zodiacus Games          Zodiacus Games   

                                              Categories  \
19290                   Single-player,Steam Achievements   
55101                                      Single-player   
34096                                      Single-player   
57398       Single-player,Steam Achievements,Steam Cloud   
26277  Single-player,Steam Achievements,Captions avai...   
44196  Single-player,Steam Achievements,Full controll...   
13127  Single-player,Steam Achievements,Full controll...   
32871                Single-player,Includes level editor   
33790                                      Single-player   
59145       Single-player,Steam Achievements,Steam Cloud   

                              Genres  \
19290  Action,Adventure,Casual,Indie 

In [38]:
games.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 52440 entries, 1 to 67973
Data columns (total 18 columns):
 #   Column                    Non-Null Count  Dtype         
---  ------                    --------------  -----         
 0   AppID                     52440 non-null  int64         
 1   Name                      52440 non-null  object        
 2   Release date              52440 non-null  datetime64[ns]
 3   Estimated owners          52440 non-null  object        
 4   Peak CCU                  52440 non-null  int64         
 5   Price                     52440 non-null  float64       
 6   Metacritic score          52440 non-null  int64         
 7   User score                52440 non-null  int64         
 8   Positive                  52440 non-null  int64         
 9   Negative                  52440 non-null  int64         
 10  Recommendations           52440 non-null  int64         
 11  Average playtime forever  52440 non-null  int64         
 12  Median playtime fo

In [39]:
games.head(10)

AppID                       Name Release date Estimated owners  \
1    655370               Train Bandit   2017-10-12        0 - 20000   
2   1732930               Jolt Project   2021-11-17        0 - 20000   
3   1355720                   Henosis™   2020-07-23        0 - 20000   
4   1139950      Two Weeks in Painland   2020-02-03        0 - 20000   
5   1469160             Wartune Reborn   2021-02-26   50000 - 100000   
8   1178150      MazM: Jekyll and Hyde   2020-04-02        0 - 20000   
9    320150  Deadlings: Rotten Edition   2014-11-11   50000 - 100000   
10  1026420                     WARSAW   2019-10-02    20000 - 50000   
11   485000             Cthulhu Realms   2016-07-01   50000 - 100000   
12  1620060          Clockwork Dungeon   2021-08-27        0 - 20000   

    Peak CCU  Price  Metacritic score  User score  Positive  Negative  \
1          0   0.99                 0           0        53         5   
2          0   4.99                 0           0         0         0   
3          0   5.99                 0           0         3         0   
4          0   0.00                 0           0        50         8   
5         68   0.00                 0           0        87        49   
8          1  14.99                 0           0        76         6   
9          0   3.99                 0           0       225        45   
10         5  23.99                62           0       589       212   
11         0   0.00                 0           0       147        58   
12         0   1.99                 0           0         5         0   

    Recommendations  Average playtime forever  Median playtime forever  \
1                 0                         0                        0   
2                 0                         0                        0   
3                 0                         0                        0   
4                 0                         0                        0   
5                 0                         0                        0   
8                 0                         0                        0   
9                 0                       703                      782   
10              427                        67                       93   
11                0                       224                      257   
12                0                         0                        0   

                Developers                     Publishers  \
1             Rusty Moyher                   Wild Rooster   
2            Campião Games                  Campião Games   
3        Odd Critter Games              Odd Critter Games   
4            Unusual Games                  Unusual Games   
5                    7Road                          7Road   
8            Growing Seeds                  CFK Co., Ltd.   
9           ONE MORE LEVEL                 ONE MORE LEVEL   
10          Pixelated Milk  Pixelated Milk,gaming company   
11  Wise Wizard Games, LLC         Wise Wizard Games, LLC   
12          Beardlings LLC                 Beardlings LLC   

                                           Categories  \
1   Single-player,Steam Achievements,Full controll...   
2                                       Single-player   
3               Single-player,Full controller support   
4                    Single-player,Steam Achievements   
5   Single-player,Multi-player,MMO,PvP,Online PvP,...   
8   Single-player,Steam Achievements,Full controll...   
9   Single-player,Steam Achievements,Steam Trading...   
10  Single-player,Steam Achievements,Steam Trading...   
11  Single-player,Multi-player,Shared/Split Screen...   
12                   Single-player,Steam Achievements   

                                               Genres  \
1                                        Action,Indie   
2                     Action,Adventure,Indie,Strategy   
3                              Adventure,Casual,Indie   
4                                     Adventure,Indie   
5   Adven

### COLUMNS FORMATTING AND CREATION

In [40]:
games.value_counts('Estimated owners')

Estimated owners
0 - 20000                36078
20000 - 50000             6424
50000 - 100000            3201
100000 - 200000           2178
200000 - 500000           1850
0 - 0                     1086
500000 - 1000000           777
1000000 - 2000000          433
2000000 - 5000000          291
5000000 - 10000000          71
10000000 - 20000000         31
20000000 - 50000000         16
50000000 - 100000000         3
100000000 - 200000000        1
dtype: int64

We can see that the most common range of estimated owners is the 0-20000 range. We are interested in changing this ranges to an absolute value to make some calculations. When transforming ranges into an absolute number it is common to take the lower bound, the average or the upper bound. I am going to take the average of the range. The problem with taking the average is that the most common range is the 0-20000 range, meaning that we have no differentiation between games that have no players and games that did(despite having a low number of players). I consider that it is important to have this differentiation. I am going to assume that games with a range of 0-20000 players, a 'Peak CCU' of 0, no negative, no positive reactions, no recomendations and no average playtime are games with no players. I will create a new column called 'Owners' with this assumptions. So, in this new column we will have games with 0 players based on our assumptions and the average of the range from the 'Estimated owners' column when this assumptions are not fulfilled. After this, I will remove those games with 0. Our focus is finding out what makes a game great. Anyone can upload a game to steam, so we have a lot of games. Many of those games never get a player, we are not interested in these games.

In [41]:
games[games['Estimated owners'] == 0]

Empty DataFrame
Columns: [AppID, Name, Release date, Estimated owners, Peak CCU, Price, Metacritic score, User score, Positive, Negative, Recommendations, Average playtime forever, Median playtime forever, Developers, Publishers, Categories, Genres, Tags]
Index: []

In [42]:
 
games['Owners'] =games['Estimated owners'].str.split('-').apply(lambda x: (int(x[0]) + int(x[1])) // 2)

In [43]:
def conditions(games):
    if (games['Owners'] == 10000) & (games['Peak CCU'] == 0) & (games['Positive'] == 0) & (games['Negative'] == 0) & (games['Recommendations'] == 0) & (games['Average playtime forever'] == 0):
        return 0
    else:
        return games['Owners']

games['Owners'] = games.apply(conditions, axis =1)

In [44]:
#We have created the Owners column based on the assumptions, let's remove the Estimated owners column:
games = games.drop('Estimated owners', axis =1)

In [45]:
games

AppID                          Name Release date  Peak CCU  Price  \
1       655370                  Train Bandit   2017-10-12         0   0.99   
2      1732930                  Jolt Project   2021-11-17         0   4.99   
3      1355720                      Henosis™   2020-07-23         0   5.99   
4      1139950         Two Weeks in Painland   2020-02-03         0   0.00   
5      1469160                Wartune Reborn   2021-02-26        68   0.00   
...        ...                           ...          ...       ...    ...   
65768  1284540  Yorna: Monster Girl's Secret   2020-09-18         3  19.99   
66476  1140440                OMON Simulator   2019-10-18       256   0.44   
67000   965940                       Qu-tros   2018-11-26         0   0.00   
67129   984570                  Chess Sphere   2018-12-07         0   0.00   
67973  1443650                  Dubbing Time   2020-11-17         0   0.00   

       Metacritic score  User score  Positive  Negative  Recommendations  \
1                     0           0        53         5                0   
2                     0           0         0         0                0   
3                     0           0         3         0                0   
4                     0           0        50         8                0   
5                     0           0        87        49                0   
...                 ...         ...       ...       ...              ...   
65768                 0           0         0         0                0   
66476                 0           0      1795       643             2101   
67000                 0           0        10        14                0   
67129                 0           0        40         5                0   
67973                 0           0         1         0                0   

       Average playtime forever  Median playtime forever  \
1                             0                        0   
2                             0                        0   
3                             0                        0   
4                             0                        0   
5                             0                        0   
...                         ...                      ...   
65768                         0                        0   
66476                        37                       18   
67000                         0                        0   
67129                         0                        0   
67973                         0                        0   

                        Developers  \
1                     Rusty Moyher   
2                    Campião Games   
3                Odd Critter Games   
4                    Unusual Games   
5                            7Road   
...                            ...   
65768                 Yeehaw Games   
66476  Ukrainian Bears,Hideo Milos   
67000              Betamaximus-128   
67129              betamaximus-128   
67973               Unique Records   

                                          Publishers  \
1                                       Wild Rooster   
2                                      Campião Games   
3                                  Odd Critter Games   
4                                      Unusual Games   
5                                              7Road   
...                                              ...   
65768                                   Yeehaw Games   
66476  United Nations,Putin The Best,Ukrainian Bears   
67000                                Betamaximus-128   
67129                                betamaximus-128   
67973                                 Unique Records   

                                              Categories  \
1      Single-player,Steam Achievements,Full controll...   
2                                          Single-player   
3                  Single-player,Full controller support   
4                       Single-player,Steam Achievements   
5      S

In [46]:
games = games[games['Owners'] != 0]
games

AppID                          Name Release date  Peak CCU  Price  \
1       655370                  Train Bandit   2017-10-12         0   0.99   
3      1355720                      Henosis™   2020-07-23         0   5.99   
4      1139950         Two Weeks in Painland   2020-02-03         0   0.00   
5      1469160                Wartune Reborn   2021-02-26        68   0.00   
8      1178150         MazM: Jekyll and Hyde   2020-04-02         1  14.99   
...        ...                           ...          ...       ...    ...   
65768  1284540  Yorna: Monster Girl's Secret   2020-09-18         3  19.99   
66476  1140440                OMON Simulator   2019-10-18       256   0.44   
67000   965940                       Qu-tros   2018-11-26         0   0.00   
67129   984570                  Chess Sphere   2018-12-07         0   0.00   
67973  1443650                  Dubbing Time   2020-11-17         0   0.00   

       Metacritic score  User score  Positive  Negative  Recommendations  \
1                     0           0        53         5                0   
3                     0           0         3         0                0   
4                     0           0        50         8                0   
5                     0           0        87        49                0   
8                     0           0        76         6                0   
...                 ...         ...       ...       ...              ...   
65768                 0           0         0         0                0   
66476                 0           0      1795       643             2101   
67000                 0           0        10        14                0   
67129                 0           0        40         5                0   
67973                 0           0         1         0                0   

       Average playtime forever  Median playtime forever  \
1                             0                        0   
3                             0                        0   
4                             0                        0   
5                             0                        0   
8                             0                        0   
...                         ...                      ...   
65768                         0                        0   
66476                        37                       18   
67000                         0                        0   
67129                         0                        0   
67973                         0                        0   

                        Developers  \
1                     Rusty Moyher   
3                Odd Critter Games   
4                    Unusual Games   
5                            7Road   
8                    Growing Seeds   
...                            ...   
65768                 Yeehaw Games   
66476  Ukrainian Bears,Hideo Milos   
67000              Betamaximus-128   
67129              betamaximus-128   
67973               Unique Records   

                                          Publishers  \
1                                       Wild Rooster   
3                                  Odd Critter Games   
4                                      Unusual Games   
5                                              7Road   
8                                      CFK Co., Ltd.   
...                                              ...   
65768                                   Yeehaw Games   
66476  United Nations,Putin The Best,Ukrainian Bears   
67000                                Betamaximus-128   
67129                                betamaximus-128   
67973                                 Unique Records   

                                              Categories  \
1      Single-player,Steam Achievements,Full controll...   
3                  Single-player,Full controller support   
4                       Single-player,Steam Achievements   
5      Single-player,Multi-player,MMO,PvP,Online PvP,...   
8      S

Let's create a new column 'Releaser year'. We extract the year from the Release date column

In [47]:

import datetime
games['Release Year'] = games['Release date'].dt.year


C:\Users\manum\AppData\Local\Temp\ipykernel_4496\1470786746.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  games['Release Year'] = games['Release date'].dt.year


Steam has a rating algorithm based on the negative and positive reviews the game has(link to the steam article:https://steamdb.info/blog/steamdb-rating/). We are going to use the steam rating formula to create a new column. 


In [48]:
import math
def rating(games):
    
    pos = games['Positive']
    neg = games['Negative']
    total_reviews = pos+neg
    
    if total_reviews > 0:
        average = pos/total_reviews

        score = average - (average*0.5) * 2**(-math.log10(total_reviews+1))

        return score *100
    else:
        return 0

games['total_reviews'] = games['Positive'] + games['Negative']
games['rating_ratio'] = games['Positive'] / games['Negative']
games['Rating'] = games.apply(rating, axis=1)



C:\Users\manum\AppData\Local\Temp\ipykernel_4496\419609567.py:19: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  games['total_reviews'] = games['Positive'] + games['Negative']
C:\Users\manum\AppData\Local\Temp\ipykernel_4496\419609567.py:20: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  games['rating_ratio'] = games['Positive'] / games['Negative']
C:\Users\manum\AppData\Local\Temp\ipykernel_4496\419609567.py:21: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try u

In [49]:
games

AppID                          Name Release date  Peak CCU  Price  \
1       655370                  Train Bandit   2017-10-12         0   0.99   
3      1355720                      Henosis™   2020-07-23         0   5.99   
4      1139950         Two Weeks in Painland   2020-02-03         0   0.00   
5      1469160                Wartune Reborn   2021-02-26        68   0.00   
8      1178150         MazM: Jekyll and Hyde   2020-04-02         1  14.99   
...        ...                           ...          ...       ...    ...   
65768  1284540  Yorna: Monster Girl's Secret   2020-09-18         3  19.99   
66476  1140440                OMON Simulator   2019-10-18       256   0.44   
67000   965940                       Qu-tros   2018-11-26         0   0.00   
67129   984570                  Chess Sphere   2018-12-07         0   0.00   
67973  1443650                  Dubbing Time   2020-11-17         0   0.00   

       Metacritic score  User score  Positive  Negative  Recommendations  ...  \
1                     0           0        53         5                0  ...   
3                     0           0         3         0                0  ...   
4                     0           0        50         8                0  ...   
5                     0           0        87        49                0  ...   
8                     0           0        76         6                0  ...   
...                 ...         ...       ...       ...              ...  ...   
65768                 0           0         0         0                0  ...   
66476                 0           0      1795       643             2101  ...   
67000                 0           0        10        14                0  ...   
67129                 0           0        40         5                0  ...   
67973                 0           0         1         0                0  ...   

                        Developers  \
1                     Rusty Moyher   
3                Odd Critter Games   
4                    Unusual Games   
5                            7Road   
8                    Growing Seeds   
...                            ...   
65768                 Yeehaw Games   
66476  Ukrainian Bears,Hideo Milos   
67000              Betamaximus-128   
67129              betamaximus-128   
67973               Unique Records   

                                          Publishers  \
1                                       Wild Rooster   
3                                  Odd Critter Games   
4                                      Unusual Games   
5                                              7Road   
8                                      CFK Co., Ltd.   
...                                              ...   
65768                                   Yeehaw Games   
66476  United Nations,Putin The Best,Ukrainian Bears   
67000                                Betamaximus-128   
67129                                betamaximus-128   
67973                                 Unique Records   

                                              Categories  \
1      Single-player,Steam Achievements,Full controll...   
3                  Single-player,Full controller support   
4                       Single-player,Steam Achievements   
5      Single-player,Multi-player,MMO,PvP,Online PvP,...   
8      Single-player,Steam Achievements,Full controll...   
...                                                  ...   
65768  Single-player,Steam Achievements,Steam Trading...   
66476  Single-player,Steam Achievements,Steam Trading...   
67000  Single-player,Partial Controller Support,Steam...   
67129  Multi-player,PvP,Online PvP,Shared/Split Scree...   
67973                                      Single-player   

                                                  Genres  \
1                                           Action,Indie   
3                                 Adventure,Casual,Indie   
4                                        Adventure,Indie   
5      A

We are going to create a column based on the price. We are going to distinguish 4 groups: free-to-play(price=0), cheap(0<x<=10), moderate(10<x<=40) & expensive(>40)

In [50]:

def price_group(games):
    if games['Price'] == 0.00:
        return 'Free-to-Play'
    elif (games['Price'] >0.00) & (games['Price'] <=10.00 ):
        return 'Cheap'
    elif (games['Price'] >10.00) & (games['Price'] <= 40.00):
        return 'Moderate'
    else:
        return 'Expensive'
    

games['Price Group'] = games.apply(price_group, axis =1)

C:\Users\manum\AppData\Local\Temp\ipykernel_4496\1071969720.py:13: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  games['Price Group'] = games.apply(price_group, axis =1)


In [51]:
games.sample(10)

AppID                                 Name Release date  Peak CCU  \
10194  1504110                            Grandiose   2021-07-07         0   
48509  1697680                     Fear Corporation   2021-08-19         0   
32605   654500                            Super XYX   2020-08-10         0   
12618   273030               Pro Rugby Manager 2015   2014-09-18         0   
12254   284390  The Last Door - Collector's Edition   2014-05-20         3   
23458  1389410      Record of Battle 3D Maiden Wars   2020-08-20         0   
16655   739990                          Lucky Panda   2017-11-07         0   
8311    601020       Seed Of The Arcane , Episode 1   2017-05-11         0   
18286   697680                          Marblesared   2017-10-04         0   
50328   773320                      TcNo TimeKeeper   2018-01-15         1   

       Price  Metacritic score  User score  Positive  Negative  \
10194   3.99                 0           0        30        26   
48509   9.99                 0           0         7         8   
32605   9.99                 0           0        66         5   
12618   9.99                 0           0        33       126   
12254   9.99                79           0      1878       105   
23458   0.00                 0           0         3         1   
16655  16.99                 0           0        49        58   
8311   12.99                 0           0        12        26   
18286   0.99                 0           0         4         0   
50328   0.00                 0           0         1         1   

       Recommendations  ...                 Publishers  \
10194                0  ...            Valentin Supiot   
48509                0  ...             PimpGameStudio   
32605                0  ...         Team Grybanser Fox   
12618              123  ...                  505 Games   
12254             1442  ...  Phoenix Online Publishing   
23458                0  ...                     Komodo   
16655                0  ...                    OnBlind   
8311                 0  ...              Infosuccess3d   
18286                0  ...                    PCAndVR   
50328                0  ...              Wesley Pyburn   

                                              Categories  \
10194      Single-player,Multi-player,Co-op,Online Co-op   
48509                                      Single-player   
32605              Single-player,Full controller support   
12618  Single-player,Steam Achievements,Captions avai...   
12254  Single-player,Steam Achievements,Steam Trading...   
23458           Single-player,Partial Controller Support   
16655                                      Single-player   
8311   Single-player,Steam Achievements,Steam Trading...   
18286           Single-player,Partial Controller Support   
50328                                        Steam Cloud   

                                                  Genres  \
10194                                              Indie   
48509                                    Adventure,Indie   
32605                                       Action,Indie   
12618                         Simulation,Sports,Strategy   
12254                                    Adventure,Indie   
23458                                   Free to Play,RPG   
16655                             Adventure,Casual,Indie   
8311   Nudity,Violent,Gore,Adventure,Indie,RPG,Early ...   
18286                                Casual,Early Access   
50328                                   Video Production   

                                                    Tags  Owners Release Year  \
10194  Strategy,Management,Survival,First-Person,Agri...   10000         2021   
48509  Horror,Survival Horror,Alternate History,Psych...   10000         2021   
32605  Action,Indie,Retro,Shoot 'Em Up,Arcade,Bullet ...   10000         2020   
12618                         Sports,Strategy,Simulation   35000         2014   
12254  Pixel Graphics,Point & Click,Story Rich,Horror... 

In the Categories and Genres columns, we can have more than one category and genre for each game. Each of these genres or categories are separated by a coma. First of all I am going to look at all the unique categories and genres we have. 


In [52]:
def categories_clean(games):
    games = games.split(",")
    return games

In [53]:

categories_count = {}
for categories in games['Categories']:
    words = categories_clean(categories)
    for word in words:
        categories_count[word] = categories_count.get(word, 0) + 1

In [54]:
categories_count

{'Single-player': 45546,
 'Steam Achievements': 24495,
 'Full controller support': 10400,
 'Steam Leaderboards': 5109,
 'Remote Play on Phone': 691,
 'Remote Play on Tablet': 845,
 'Remote Play on TV': 1893,
 'Multi-player': 10013,
 'MMO': 693,
 'PvP': 6270,
 'Online PvP': 4419,
 'Co-op': 4898,
 'Online Co-op': 2513,
 'In-App Purchases': 1231,
 'Steam Trading Cards': 9123,
 'Partial Controller Support': 7091,
 'Steam Cloud': 12637,
 'Shared/Split Screen': 3884,
 'Cross-Platform Multiplayer': 1573,
 'Remote Play Together': 4448,
 'Stats': 2717,
 'Shared/Split Screen PvP': 2863,
 'no Category added': 828,
 'Captions available': 976,
 'Steam Workshop': 1493,
 'Includes level editor': 1507,
 'LAN PvP': 369,
 'LAN Co-op': 327,
 'Shared/Split Screen Co-op': 2209,
 'Steam Turn Notifications': 78,
 'VR Support': 239,
 'SteamVR Collectibles': 41,
 'Valve Anti-Cheat enabled': 94,
 'Includes Source SDK': 33,
 'Commentary available': 196,
 'Mods': 1}

In [55]:
def genres_clean(games):
    games = games.split(',')
    return games

In [56]:
genres_count = {}
for genres in games['Genres']:
    words = genres_clean(genres)
    for word in words:
        genres_count[word] = genres_count.get(word, 0) + 1

In [57]:
genres_count

{'Action': 20747,
 'Indie': 35194,
 'Adventure': 18832,
 'Casual': 19547,
 'Free to Play': 3337,
 'Massively Multiplayer': 1266,
 'RPG': 8260,
 'Strategy': 9396,
 'Simulation': 9514,
 'Early Access': 5054,
 'Sports': 2316,
 'Racing': 1794,
 'Utilities': 595,
 'Education': 280,
 'Sexual Content': 104,
 'Nudity': 115,
 'Violent': 495,
 'Gore': 298,
 'Animation & Modeling': 282,
 'Design & Illustration': 368,
 'Game Development': 130,
 'Web Publishing': 83,
 'Software Training': 143,
 'Photo Editing': 95,
 'Audio Production': 152,
 'Video Production': 202,
 'no Genres added': 60,
 'Accounting': 17,
 'Movie': 2,
 'Documentary': 1,
 'Episodic': 1,
 'Short': 1,
 'Tutorial': 1,
 '360 Video': 1}

We are going to select just the single player and multiplayer Categories. Most games have one of these categories and the ones that  do not, are not a game or are a simulator test. The categories that are more specific like MMO also appear in the Genre column. In the Genre column we are going to select only the game genres. We are going to remove Genres like Tutorials, Design, Software Training... I am going to create a column for each Category and Genre we want in our dataset. Each of these columns are going to be a dummy column. They are going to have a 1 if the game has the genre of the column and a 0 if that is not the case. For example, imagine we have a game with category: Multiplayer and genre: action, adventure. We will have many columns with the different categories and genres. This game will have  a 1 in the multiplayer, action and adventure columns. In the rest of columns will have a 0. Once we have done this process, I will remove all those games that do not have a 1 in any column. This games will be those games that I do not consider games. They have genres like simulator, test, tutorial...


        


In [60]:
games['Multi_player'] = games['Categories'].apply(lambda x: 1 if 'Multi-player' in x.split(',') else 0)
games['Single_player'] = games['Categories'].apply(lambda x: 1 if 'Single-player' in x.split(',') else 0)
games['Action'] = games['Genres'].apply(lambda x: 1 if 'Action' in x.split(',') else 0)
games['Adventure'] = games['Genres'].apply(lambda x: 1 if 'Adventure' in x.split(',') else 0)
games['Casual'] = games['Genres'].apply(lambda x: 1 if 'Casual' in x.split(',') else 0)
games['Indie'] = games['Genres'].apply(lambda x: 1 if 'Indie' in x.split(',') else 0)
games['Massively Multiplayer'] = games['Genres'].apply(lambda x: 1 if 'Massively Multiplayer' in x.split(',') else 0)
games['RPG'] = games['Genres'].apply(lambda x: 1 if 'RPG' in x.split(',') else 0)
games['Racing'] = games['Genres'].apply(lambda x: 1 if 'Racing' in x.split(',') else 0)
games['Simulation'] = games['Genres'].apply(lambda x: 1 if 'Simulation' in x.split(',') else 0)
games['Strategy'] = games['Genres'].apply(lambda x: 1 if 'Strategy' in x.split(',') else 0)
games['Sports'] = games['Genres'].apply(lambda x: 1 if 'Sports' in x.split(',') else 0)

C:\Users\manum\AppData\Local\Temp\ipykernel_4496\3361676039.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  games['Multi_player'] = games['Categories'].apply(lambda x: 1 if 'Multi-player' in x.split(',') else 0)
C:\Users\manum\AppData\Local\Temp\ipykernel_4496\3361676039.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  games['Single_player'] = games['Categories'].apply(lambda x: 1 if 'Single-player' in x.split(',') else 0)
C:\Users\manum\AppData\Local\Temp\ipykernel_4496\3361676039.py:3: SettingWithC

In [61]:
games.columns

Index(['AppID', 'Name', 'Release date', 'Peak CCU', 'Price',
       'Metacritic score', 'User score', 'Positive', 'Negative',
       'Recommendations', 'Average playtime forever',
       'Median playtime forever', 'Developers', 'Publishers', 'Categories',
       'Genres', 'Tags', 'Owners', 'Release Year', 'total_reviews',
       'rating_ratio', 'Rating', 'Price Group', 'Multi_player',
       'Single_player', 'Action', 'Adventure', 'Casual', 'Indie',
       'Massively Multiplayer', 'RPG', 'Racing', 'Simulation', 'Strategy',
       'Sports'],
      dtype='object')

In [66]:
games.iloc[:,25:]

Action  Adventure  Casual  Indie  Massively Multiplayer  RPG  Racing  \
1           1          0       0      1                      0    0       0   
3           0          1       1      1                      0    0       0   
4           0          1       0      1                      0    0       0   
5           0          1       1      0                      1    1       0   
8           0          1       0      0                      0    1       0   
...       ...        ...     ...    ...                    ...  ...     ...   
65768       1          1       1      1                      0    1       0   
66476       1          1       0      1                      0    1       0   
67000       0          0       1      0                      0    0       0   
67129       0          0       1      0                      0    0       0   
67973       1          0       1      1                      0    0       0   

       Simulation  Strategy  Sports  
1               0         0       0  
3               0         0       0  
4               0         0       0  
5               0         1       0  
8               1         1       0  
...           ...       ...     ...  
65768           0         0       0  
66476           1         1       0  
67000           0         1       0  
67129           0         1       0  
67973           1         0       0  

[48682 rows x 10 columns]

In [67]:
games['genre_sum'] = games.iloc[:,25:].sum(axis=1)

C:\Users\manum\AppData\Local\Temp\ipykernel_4496\163224511.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  games['genre_sum'] = games.iloc[:,25:].sum(axis=1)


In [68]:
games = games[games['genre_sum'] != 0]
games

AppID                          Name Release date  Peak CCU  Price  \
1       655370                  Train Bandit   2017-10-12         0   0.99   
3      1355720                      Henosis™   2020-07-23         0   5.99   
4      1139950         Two Weeks in Painland   2020-02-03         0   0.00   
5      1469160                Wartune Reborn   2021-02-26        68   0.00   
8      1178150         MazM: Jekyll and Hyde   2020-04-02         1  14.99   
...        ...                           ...          ...       ...    ...   
65768  1284540  Yorna: Monster Girl's Secret   2020-09-18         3  19.99   
66476  1140440                OMON Simulator   2019-10-18       256   0.44   
67000   965940                       Qu-tros   2018-11-26         0   0.00   
67129   984570                  Chess Sphere   2018-12-07         0   0.00   
67973  1443650                  Dubbing Time   2020-11-17         0   0.00   

       Metacritic score  User score  Positive  Negative  Recommendations  ...  \
1                     0           0        53         5                0  ...   
3                     0           0         3         0                0  ...   
4                     0           0        50         8                0  ...   
5                     0           0        87        49                0  ...   
8                     0           0        76         6                0  ...   
...                 ...         ...       ...       ...              ...  ...   
65768                 0           0         0         0                0  ...   
66476                 0           0      1795       643             2101  ...   
67000                 0           0        10        14                0  ...   
67129                 0           0        40         5                0  ...   
67973                 0           0         1         0                0  ...   

       Adventure  Casual Indie Massively Multiplayer RPG Racing Simulation  \
1              0       0     1                     0   0      0          0   
3              1       1     1                     0   0      0          0   
4              1       0     1                     0   0      0          0   
5              1       1     0                     1   1      0          0   
8              1       0     0                     0   1      0          1   
...          ...     ...   ...                   ...  ..    ...        ...   
65768          1       1     1                     0   1      0          0   
66476          1       0     1                     0   1      0          1   
67000          0       1     0                     0   0      0          0   
67129          0       1     0                     0   0      0          0   
67973          0       1     1                     0   0      0          1   

       Strategy  Sports  genre_sum  
1             0       0          2  
3             0       0          3  
4             0       0          2  
5             1       0          5  
8             1       0          4  
...         ...     ...        ...  
65768         0       0          5  
66476         1       0          6  
67000         1       0          2  
67129         1       0          2  
67973         0       0          4  

[47568 rows x 36 columns]

In [69]:
games = games.drop('genre_sum', axis=1)
games

AppID                          Name Release date  Peak CCU  Price  \
1       655370                  Train Bandit   2017-10-12         0   0.99   
3      1355720                      Henosis™   2020-07-23         0   5.99   
4      1139950         Two Weeks in Painland   2020-02-03         0   0.00   
5      1469160                Wartune Reborn   2021-02-26        68   0.00   
8      1178150         MazM: Jekyll and Hyde   2020-04-02         1  14.99   
...        ...                           ...          ...       ...    ...   
65768  1284540  Yorna: Monster Girl's Secret   2020-09-18         3  19.99   
66476  1140440                OMON Simulator   2019-10-18       256   0.44   
67000   965940                       Qu-tros   2018-11-26         0   0.00   
67129   984570                  Chess Sphere   2018-12-07         0   0.00   
67973  1443650                  Dubbing Time   2020-11-17         0   0.00   

       Metacritic score  User score  Positive  Negative  Recommendations  ...  \
1                     0           0        53         5                0  ...   
3                     0           0         3         0                0  ...   
4                     0           0        50         8                0  ...   
5                     0           0        87        49                0  ...   
8                     0           0        76         6                0  ...   
...                 ...         ...       ...       ...              ...  ...   
65768                 0           0         0         0                0  ...   
66476                 0           0      1795       643             2101  ...   
67000                 0           0        10        14                0  ...   
67129                 0           0        40         5                0  ...   
67973                 0           0         1         0                0  ...   

       Action  Adventure Casual Indie Massively Multiplayer RPG Racing  \
1           1          0      0     1                     0   0      0   
3           0          1      1     1                     0   0      0   
4           0          1      0     1                     0   0      0   
5           0          1      1     0                     1   1      0   
8           0          1      0     0                     0   1      0   
...       ...        ...    ...   ...                   ...  ..    ...   
65768       1          1      1     1                     0   1      0   
66476       1          1      0     1                     0   1      0   
67000       0          0      1     0                     0   0      0   
67129       0          0      1     0                     0   0      0   
67973       1          0      1     1                     0   0      0   

       Simulation  Strategy  Sports  
1               0         0       0  
3               0         0       0  
4               0         0       0  
5               0         1       0  
8               1         1       0  
...           ...       ...     ...  
65768           0         0       0  
66476           1         1       0  
67000           0         1       0  
67129           0         1       0  
67973           1         0       0  

[47568 rows x 35 columns]

I am going to add a new column that differentiates between free and paid games. We still have a column differentiating between price ranges, but I think that this column can also be useful for a future analysis.

In [70]:
# I add a column that differentiates between free-to-play and paid games:
def price_type(games):
    if games['Price'] == 0.00:
        return 'Free'
    else:
        return 'Paid'
games['Price_type']= games.apply(price_type,axis=1)

In [71]:
games.columns

Index(['AppID', 'Name', 'Release date', 'Peak CCU', 'Price',
       'Metacritic score', 'User score', 'Positive', 'Negative',
       'Recommendations', 'Average playtime forever',
       'Median playtime forever', 'Developers', 'Publishers', 'Categories',
       'Genres', 'Tags', 'Owners', 'Release Year', 'total_reviews',
       'rating_ratio', 'Rating', 'Price Group', 'Multi_player',
       'Single_player', 'Action', 'Adventure', 'Casual', 'Indie',
       'Massively Multiplayer', 'RPG', 'Racing', 'Simulation', 'Strategy',
       'Sports', 'Price_type'],
      dtype='object')

Remove the columns I am not interested in.

In [72]:
games = games.drop(columns =['AppID','Metacritic score','Genres','Tags','User score', 'Recommendations','Median playtime forever','Categories'], axis=1)

In [73]:
games.describe()[['Price', 'Average playtime forever','Rating']]

Price  Average playtime forever        Rating
count  47568.000000              47568.000000  47568.000000
mean       7.390016                150.981059     59.166055
std       10.200826               1155.789276     21.261377
min        0.000000                  0.000000      0.000000
25%        1.000000                  0.000000     48.110901
50%        4.990000                  0.000000     64.079515
75%        9.990000                 11.000000     74.404027
max      999.000000              90351.000000     97.688275

In price we may have an outlier. Let's look how many games are more expensive than 200$. In Average playtime despite having a max value 
distant from the mean, it is normal to have games that have a lot more hours of play than others. In rating, we have no outliers.


In [75]:
games[games['Price'] > 200][['Name', 'Developers', 'Publishers','Owners','Price', 'Peak CCU', 'Positive', 'Negative']]

Name  Developers  Publishers  Owners  \
26936  Ascent Free-Roaming VR Experience  Fury Games  Fury Games   10000   

       Price  Peak CCU  Positive  Negative  
26936  999.0         0         6         0

We have some games around 250$, and one that cost 1000$. I want to look at the stats of this game to decide if we have to remove it from the dataset. If the game has low 'owners', 'peak CCU' and recomendations, we will remove it. We see low activity in the game so we remove it from the dataset:
games = games.drop(labels = 26936, axis = 0)

Get this dataset to a csv to make our analysis

games
games.to_csv('steam_cleaned2.csv')